files/default/slurm/pcluster_slurm_config_generator.py

```diff
@@ -0,0 +1,160 @@
+    """
+    Load queues_info and add information used to render templates.
+
+    :return: queues_info containing id for first queue, master_hostname and queue_name
+    """
+    with open(input_file_path) as input_file:
+        queues_info = json.load(input_file)
+        queues_info["master_hostname"] = gethostname()
+        # Mark the first queue so we can set first queue as default,
+        first = True
+        for queue in queues_info["queues_config"]:
+            if first:
+                queues_info["queues_config"][queue]["is_first_queue"] = True
+                first = False
+            queues_info["queues_config"][queue]["queue_name"] = queue
+        return queues_info
+
+
+def _render_queue_configs(queues_info, queue, file_type, env, pcluster_subdirectory):
+    """
+    Render queue-specific config based on queues_info, and sets queue filename in queues_info.
+
+    :return: rendered queue-specific templates
+    """
+    queues_info["queues_config"][queue]["queue_{}_filename".format(file_type)] = path.join(
+        pcluster_subdirectory, "slurm_parallelcluster_{}_{}.conf".format(queue, file_type)
+    )
+    return env.get_template("slurm_parallelcluster_queue_{}.conf".format(file_type)).render(
+        queue=queues_info["queues_config"][queue]
+    )
+
+
+def _render_slurm_parallelcluster_configs(queues_info, template_name, env):
+    """
+    Render slurm_parallelcluster config based on queues_info.
+
+    :return: rendered slurm_parallelcluster templates
+    """
+    return env.get_template("{}".format(template_name)).render(queues_info)
+
+
+def _get_jinja_env(template_directory):
+    """Return jinja environment with trim_blocks/lstrip_blocks set to True."""
+    file_loader = FileSystemLoader(template_directory)
+    return Environment(loader=file_loader, trim_blocks=True, lstrip_blocks=True)
+
+
+def _write_rendered_template_to_file(rendered_template, filename):
+    logger.info("Writing contents of {}".format(filename))
+    with open(filename, "w") as output_file:
+        output_file.write(rendered_template)
+
+
+def _config_logger():
+    """
+    Define a logger for config generator.
+
+    :return: the logger
+    """
+    try:
+        logfile = path.expanduser(LOG_FILE_PATH)
+        logdir = path.dirname(logfile)
+        makedirs(logdir)
+    except OSError as e:
+        if e.errno == errno.EEXIST and path.isdir(logdir):
+            pass
+        else:
+            print("Cannot create log file (%s). Failed with exception: %s" % (logfile, e))
+            exit(1)
+
+    formatter = logging.Formatter("%(asctime)s %(levelname)s [%(module)s:%(funcName)s] %(message)s")
+
+    logfile_handler = RotatingFileHandler(logfile, maxBytes=5 * 1024 * 1024, backupCount=1)
+    logfile_handler.setFormatter(formatter)
+
+    logger = logging.getLogger("pcluster_slurm_config_generator")
+    logger.addHandler(logfile_handler)
+
+    logger.setLevel("INFO")
+    return logger
+
+
+if __name__ == "__main__":
```

Comment:
nit: let's move whatever is under `if __name__ == "__main__":` to a `main()` function